src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -331,6 +346,43 @@ private HashSet<string> CollectOutputs(Logger logger)
             return (earliest, earliestPath);
         }
 
+        private bool CheckMarkers(Logger logger, IDictionary<string, DateTime> timestampCache)
+        {
+            if (!string.IsNullOrWhiteSpace(_markerFile) && _referenceMarkerFiles.Any())
+            {
+                foreach (var referenceMarkerFile in _referenceMarkerFiles)
+                {
+                    logger.Verbose("Found input marker '{0}'.", referenceMarkerFile);
+                }
+                logger.Verbose("Found output marker '{0}'.", _markerFile);
+
+                var latestInputMarker = GetLatestInput(_referenceMarkerFiles, timestampCache);
```

Comment:
Possible future improvement: you don't actually need the latest here, since the only question is "is _any_ reference newer than the output". You might want to look at [MSBuild's `IsAnyOutOfDate`](https://github.com/Microsoft/msbuild/blob/daddbee97e937a87f3cda7bb3be3405df742c57d/src/Build/BackEnd/Components/RequestBuilder/TargetUpToDateChecker.cs#L984) for ideas.